app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.tsx

```diff
@@ -97,9 +97,26 @@ export const FormBranches = ({
 
   const clearSubmitErrors = () => dispatch({ type: "clearSubmitErrors" });
 
+  // We show two branches by default, an empty control/reference branch and treatment
+  // branch, which are given a default branch name. If a branch slug is not set, it means
+  // that branch hasn't been saved by the user. If only a reference branch was previously
+  // saved we _don't_ show a new treatment branch by default and instead, if "handleAddBranch"
+  // is called, we show the empty treatment branch instead of creating a new one.
+  const [showFirstTreatmentBranch, setShowFirstTreatmentBranch] = useState(
+    !defaultValues.referenceBranch?.slug ||
+      (defaultValues.treatmentBranches &&
+        defaultValues.treatmentBranches[0]?.slug),
+  );
+
   const handleAddBranch = () => {
     commitFormData();
-    dispatch({ type: "addBranch" });
+    if (
+      showFirstTreatmentBranch ||
+      (!showFirstTreatmentBranch && defaultValues.treatmentBranches === null)
```

Comment:
I don't think in practical use, `treatmentBranches` is ever actually `null`, but our types tell us it can be and we have it set that way in some tests/stories.